automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala

```diff
@@ -661,5 +662,15 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
     }
   }
 
+  def verifyNotebookHeaders(url: String, headerName: String, headerValue: Option[String] = None)(implicit token: AuthToken): Unit ={
```

Comment:
This method confuses me. You could probably just verify the headers directly in the test -- I made a suggestion below.